src/features/npm-cdn-link.js

```diff
@@ -0,0 +1,148 @@
+let currentPageUrl = '';
+
+function hasRouteChanged () {
+	const nextPageUrl = window.location.href;
+
+	if (currentPageUrl === '' || currentPageUrl !== nextPageUrl) {
+		currentPageUrl = nextPageUrl;
+		return true;
+	}
+
+	return false;
+}
+
+function removeCDNLink () {
+	const existingLink = document.querySelector('#jsd-npm-container');
+
+	if (existingLink) {
+		existingLink.remove();
+	}
+}
+
+function setCDNLink ({ name, cdnUrl }) {
+	const headingElements = [ ...document.querySelectorAll('h3') ];
+	const installHeadingElement = headingElements.find(element => element.innerText.toLowerCase() === 'install');
+
+	if (!installHeadingElement) {
+		return;
+	}
+
+	function handleLinkClick () {
+		copy(cdnUrl);
+		setToast({
+			message: 'Copied to clipboard!',
+		});
+	}
+
+	const link = (
+		<div id="jsd-npm-container">
+			<h3 class="jsd-npm-title">jsdelivr cdn</h3>
+			{cdnUrl
+				? (
+					<div class="jsd-npm-link-container">
+						<input
+							class="jsd-npm-link-input"
+							value={cdnUrl}
+							onClick={handleLinkClick}
+							readonly
+						/>
+					</div>
+				)
+				: (
+					<div class="jsd-npm-no-file-container">
+						<p class="jsd-npm-no-file-message">
+							Sadly, this package doesn't have a default file set.
+						</p>
+						<a
+							class="jsd-npm-button"
+							href={`https://www.jsdelivr.com/package/npm/${name}`}
+							target="_blank"
+							rel="nofollow"
+						>
+							Open in jsDelivr
+						</a>
+					</div>
+				)
+			}
+		</div>
+	);
+
+	const installCommandElement = installHeadingElement.nextElementSibling;
+
+	if (installCommandElement) {
+		installCommandElement.insertAdjacentElement('afterend', link);
+		currentPageUrl = window.location.href;
+	}
+}
+
+function removeToast () {
+	const toastElement = document.querySelector('#jsd-npm-toast');
+
+	if (toastElement) {
+		toastElement.remove();
+	}
+}
+
+function setToast ({ message = '' }) {
+	const toast = (
+		<div
+			id="jsd-npm-toast"
+			class="jsd-npm-toast-container"
+			onClick={removeToast}
+		>
+			<p class="jsd-npm-toast-message">{message}</p>
+			<p class="jsd-npm-toast-close-btn">×</p>
+		</div>
+	);
+
+	document.querySelector('body').append(toast);
+	setTimeout(() => {
+		removeToast();
+	}, 2000);
+}
+
+chrome.runtime.onMessage.addListener(async (request) => {
+	const { type } = request;
+
+	if (type === 'TAB_UPDATE' && hasRouteChanged() && isPackageRoot()) {
+		removeCDNLink();
+
+		const { name, version: versionName } = getPackageDetails();
+
+		if (!name) {
+			return;
+		}
+
+		const version = versionName ? versionName : await getPackageVersionMemo(name, 'master');
+
+		if (!version) {
+			return;
+		}
+
+		const defaultFile = await getDefaultFileMemo(name, version);
+
+		// Validate file name, since `bootstrap`
+		// has default file name without `.js` extension
+		const cdnUrl = defaultFile && hasJSExtension(defaultFile) ? `https://cdn.jsdelivr.net/npm/${name}@${version}${defaultFile}` : '';
```

Comment:
It would be better to check if the default file exists in the package, regardless of its name/extension.

Reply:
We need to check for extension as well, because Bootstrap has a default file set but it is missing `.js` extension.
[Here](https://data.jsdelivr.com/v1/package/npm/bootstrap@4.3.1) is the Bootstrap's API response.

Reply:
I know but in that case, you'll just find that there's no file called `/dist/js/bootstrap` in the list. Especially if you request https://data.jsdelivr.com/v1/package/npm/bootstrap@4.3.1/flat instead, then you just do `.find()` on all files. Found => ok, show link; not found => show error.

Reply:
This is also a good approach; will make the changes.

Reply:
Finding default file in a list breaks when the API returns a minified version of it.

Example: React's package has `/index.js` file but API is returning its minified file - `/index.min.js`, which is not available in its list.
API - [https://data.jsdelivr.com/v1/package/npm/react@16.10.2/flat](https://data.jsdelivr.com/v1/package/npm/react@16.10.2/flat)

Reply:
Indeed, the check will also need to fall back to the non-minified version:

```js
getNonMinifiedName (name) {
	return name.replace(/\.min\.(js|css)$/i, '.$1');
}
```

Reply:
It's working now with fallback to non-minified version.

But I have 2 things on my mind:
1. Can't we assume that the file in `default` property exists in the package?

2. I think we should still check for file extension first and then check whether it exists in the package.
Reason: For a large (and popular) package like Bootstrap, whose default file is `/dist/js/bootstrap`; if we are not checking for file extension then we will have to compare it against 151 files even though we know it is not available.

Reply:
1. Strictly speaking, it doesn't have to exist. But I wanted to do it this way because it also solves the extension problem.
2. The problem is there are many valid extensions, not just `.js`. We could check that the file has *some* extension, as an extension-less file doesn't make much sense, but again, strictly speaking, the extension is required.

Reply:
Ok. Let's find the file from package list itself.

Reply:
Updated the pull request with above discussed changes.